algtest/perf_util.c

```diff
@@ -44,32 +45,36 @@ TPM2_RC sign(
                 &p1,
                 &s2,
                 &y2,
-                noncePoint ? noncePoint : &K, // BUG? The rG value seems to be stored here instead of E
+                &K,
                 &L,
-                &E,
+                noncePoint ? noncePoint : &E, // BUG - Some TPMs output the rG value in K instead of E
                 &inSchemeCopy.details.ecdaa.count,
                 &rspAuthsArray);
 
-        if(L.size > 4) {
-            printf("Unexpected point output in TPM2_Commit: L 04");
-            for(int i = 0; i < L.point.x.size; ++i) {
-                printf("%02x", L.point.x.buffer[i]);
+        if(K.size > 4 && K.point.x.size < 512 && K.point.y.size < 512) {
+            char point[1024];
+            char* ptr = point;
+            ptr += sprintf(ptr, "04");
+            for(int i = 0; i < K.point.x.size; ++i) {
+                ptr += sprintf(ptr, "%02x", K.point.x.buffer[i]);
             }
-            for(int i = 0; i < L.point.y.size; ++i) {
-                printf("%02x", L.point.y.buffer[i]);
+            for(int i = 0; i < K.point.y.size; ++i) {
+                ptr += sprintf(ptr, "%02x", K.point.y.buffer[i]);
```

Comment:
if sprintf fail, it returns negative value. Perhaps just check that return value is always 2 and fail otherwise...